org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/internal/core/manipulation/StubUtility.java

```diff
@@ -140,7 +140,13 @@ public class StubUtility {
 	 * Don't use this method directly, use CodeGeneration.
 	 */
 	public static String getMethodBodyContent(boolean isConstructor, IJavaProject project, String destTypeName, String methodName, String bodyStatement, String lineDelimiter) throws CoreException {
-		String templateName= isConstructor ? CodeTemplateContextType.CONSTRUCTORSTUB_ID : CodeTemplateContextType.METHODSTUB_ID;
+		return getMethodBodyContent(false, isConstructor, project, destTypeName, methodName, bodyStatement, lineDelimiter);
+	}
+
+	public static String getMethodBodyContent(boolean isSuperMethodInvocation, boolean isConstructor, IJavaProject project, String destTypeName, String methodName, String bodyStatement, String lineDelimiter) throws CoreException {
+		String templateName= isConstructor ? CodeTemplateContextType.CONSTRUCTORSTUB_ID
+							: isSuperMethodInvocation ? CodeTemplateContextType.METHODSTUB_SUPER_ID
```

Comment:
This doesn't really represent `isSuperMethodInvocation`. It's more like preserve-the-old-method-stub-for-parent-methods-when-the-stub-changes . I think the approach of creating a second template will probably work, but I'm wondering if we can avoid it.

I need to think about if there's another way to do this. I was thinking of the fact that each template has a context type, and each context type defines a set of variables that it can resolve. What if we just added another variable to the context type for the method body that represents the unsupported operation string. Between JDT-LS & JDT, we can have different templates, so maybe there's a way to just resolve that variable in both cases.

eg. If we set the "unsupported operation variable" here then evaluate the template as needed with it. Might need to discuss this further.

Reply:
Ok I've come to see this as probably the best approach for now, but I still think we should rename the parameter because it's a lot more general than the use JDT-LS would have for it.